hab-pkg-deb/bin/hab-pkg-deb.sh

```diff
@@ -0,0 +1,390 @@
+        ;;
+      *)
+        printf -- "ERROR: $1\n"
+        ;;
+    esac
+  fi
+  exit $2
+}
+
+# **Internal** Print a warning line on stderr. Takes the rest of the line as its
+# only argument.
+#
+# ```sh
+# warn "Checksum failed"
+# ```
+warn() {
+  case "${TERM:-}" in
+    *term | xterm-* | rxvt | screen | screen-*)
+      printf -- "\033[1;33mWARN: \033[1;37m$1\033[0m\n" >&2
+      ;;
+    *)
+      printf -- "WARN: $1\n" >&2
+      ;;
+  esac
+}
+
+find_system_commands() {
+  wtf=$(mktemp --version)
+  if $(mktemp --version 2>&1 | grep -q 'GNU coreutils'); then
+    _mktemp_cmd=$(command -v mktemp)
+  else
+    if $(/bin/mktemp --version 2>&1 | grep -q 'GNU coreutils'); then
+      _mktemp_cmd=/bin/mktemp
+    else
+      exit_with "We require GNU mktemp to build applications archives; aborting" 1
+    fi
+  fi
+}
+
+# The size of the package when installed.
+#
+# Per http://www.debian.org/doc/debian-policy/ch-controlfields.html, the
+# disk space is given as the integer value of the estimated installed
+# size in bytes, divided by 1024 and rounded up.
+installed_size() {
+  du "$deb_context" --apparent-size --block-size=1024 --summarize | cut -f1
+}
+
+# The package priority.
+#
+# Can be one of required, important, standard, optional, or extra.
+# See https://www.debian.org/doc/manuals/debian-faq/ch-pkg_basics.en.html#s-priority
+#
+# TODO: Allow customizing this
+priority() {
+  echo extra
+}
+
+# The package section.
+#
+# See https://www.debian.org/doc/debian-policy/ch-archive.html#s-subsections
+#
+# TODO: Allow customizing this
+section() {
+  echo misc
+}
+
+# parse the CLI flags and options
+parse_options() {
+  echo "@=$@"
+  opts="$(getopt \
+    --longoptions help,version,preinst:,postinst:,prerm:,postrm:,conflicts:,provides:,replaces: \
+    --name "$program" --options h,V -- "$@" \
+  )"
+  eval set -- "$opts"
+
+  while :; do
+    case "$1" in
+      -h | --help)
+        print_help
+        exit
+        ;;
+      -V | --version)
+        echo "$program $version"
+        exit
+        ;;
+      --preinst)
+        preinst=$2
+        shift 2
+        ;;
+      --postinst)
+        postinst=$2
+        shift 2
+        ;;
+      --prerm)
+        prerm=$2;
+        shift 2
+        ;;
+      --postrm)
+        postrm=$2
+        shift 2
+        ;;
+      --conflicts)
+        conflicts=$2
+        shift 2
+        ;;
+      --provides)
+        provides=$2
+        shift 2
+        ;;
+      --replaces)
+        replaces=$2
+        shift 2
+        ;;
+      --)
+        shift
+        pkg=$*
+        break
+        ;;
+      *)
+        exit_with "Unknown error" 1
+        ;;
+    esac
+  done
+  if [ -z "$pkg" ] || [ "$pkg" = "--" ]; then
+    print_help
+    exit_with "You must specify a Habitat package." 1
+  fi
+}
+
+# The name converted to all lowercase to be compatible with Debian naming
+# conventions
+convert_name() {
+  safe_name="${pkg_origin,,}-${pkg_name,,}"
```

Comment:
This lowercasing variable expansion is only available in Bash 4.0+. This shouldn't be an issue as we control the Bash version but we might want to leave a comment.